src/validation.cpp

```diff
@@ -4028,6 +4028,16 @@ static bool ContextualCheckBlockHeader(const CBlockHeader& block, BlockValidatio
     if (block.GetBlockTime() <= pindexPrev->GetMedianTimePast())
         return state.Invalid(BlockValidationResult::BLOCK_INVALID_HEADER, "time-too-old", "block's timestamp is too early");
 
+    // Testnet4 only: Check timestamp against prev for difficulty-adjustment
```

Comment:
06c2c713c52b60231efc3e00d2c5eb0bf9e345f9: I initially thought that this should be moved to `ConnectBlock`, somewhere in the "Sanity checks", maybe right before `bool fScriptChecks = true;`. But it's fine here.

It doesn't _need_ to be here, because we don't use the system clock for this check. However it's more readable to have it right next to the `time-too-old` check. So is that safe?

`ContextualCheckBlockHeader` is only called when we receive a header. It's called by `AcceptBlockHeader` (and by the miner), which in turn is called by `ProcessNewBlockHeaders` (header message or compact block) and `AcceptBlock` (block message, to store on disk). Notably it's _not_ called during a reorg or when doing a reindex. However in both these cases the headers will have been previously checked for  this issue.

Reply:
I will keep it here for now unless other reviewers think it should be moved.